main.tf

```
@@ -2538,8 +2538,20 @@ data "aws_iam_policy_document" "fargate_fluentbit" {
       resources = var.fargate_fluentbit.s3_bucket_arns
     }
   }
-}
 
+  dynamic "statement" {
+    for_each = length(lookup(var.fargate_fluentbit, "data_firehose_arns", [])) > 0 ? [1] : []
+
+    content {
+      sid = "FirehoseEvent"
+      actions = [
+         "firehose:PutRecord",
+          "firehose:PutRecordBatch"
+      ]
+      resources = var.fargate_fluentbit.data_firehose_arns
```

this has the chance to fail when ARNs are not provided so we should guard against that:
```suggestion
      resources = lookup(var.fargate_fluentbit, "firehose_arns", []) 
```